src/main/java/org/broadinstitute/dsm/model/PDF/MiscPDFDownload.java

```diff
@@ -0,0 +1,70 @@
+package org.broadinstitute.dsm.model.PDF;
+
+import lombok.NonNull;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.util.ElasticSearchUtil;
+import org.broadinstitute.dsm.util.ParticipantUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class MiscPDFDownload {
+    Logger logger = LoggerFactory.getLogger(MiscPDFDownload.class);
+    public Object returnPDFS(@NonNull String ddpParticipantId, String realm) {
+        DDPInstance instance = DDPInstance.getDDPInstance(realm);
+        Map<String, Map<String, Object>> participantESData;
+        if (ParticipantUtil.isGuid(ddpParticipantId)) {
+            participantESData = ElasticSearchUtil.getFilteredDDPParticipantsFromES(instance,
+                    ElasticSearchUtil.BY_GUID + ddpParticipantId);
+        }
+        else {// altpid
+            participantESData = ElasticSearchUtil.getFilteredDDPParticipantsFromES(instance, ElasticSearchUtil.BY_LEGACY_ALTPID + ddpParticipantId);
+        }
+        if (participantESData != null && !participantESData.isEmpty() && participantESData.size() == 1) {
+            Map<String, Object> participantData = participantESData.get(ddpParticipantId);
+            if (participantData != null) {
+                Map<String, Object> dsm = (Map<String, Object>) participantData.get(ElasticSearchUtil.DSM);
+                if (dsm != null) {
+                    Object pdf = dsm.get(ElasticSearchUtil.PDFS);
+                    return pdf;
+                }
+            }
+            else {
+                for (Object value : participantESData.values()) {
+                    participantData = (Map<String, Object>) value;
+                    //check that it is really right participant
+                    if (participantData != null) {
```

Comment:
too many if chain, guard condition approach could provide better visibility, or at least this whole if chains could be extracted as a method with some meaningful name.